test/apps/iodemo/ucx_wrapper.h

```
@@ -68,12 +68,26 @@ class UcxLog {
 };
 
 
+/*
+ * Holds details of arrived AM message
+ */
+struct UcxAmDesc {
```

can we avoid this struct and pass only relevant fields?

---

you mean pass data and params as separate arguments? I'd better pass single parameter (as it is now), because IODEMO does nothing with it except passing to UcxWrapper recv function.

---

if it does nothing, maybe it should opaque type which is only in ucx_wrapper.cc but not in ucx_wrapper.h?